### ISSUE TITLE
Make Joi support tree-shakable

### ISSUE THREAD

**Issue by djmitche** [taskcluster/react-schema-viewer#22]
Right now, even if you don't use Joi, you end up pulling it.

Let's let tree-shaking omit that.  That means a separately importable component for the Joi schema viewer, probably just converting to json-schema and wrapping the "main" viewer component.

**Comment by arshadkazmi42** [comment]
@djmitche I can work on this. Can you provide some more details, what exaclty needs to be done here?

**Comment by djmitche** [comment]
"Tree shaking" is the process where Webpack looks at what symbols are being imported from a module, and only includes those that are imported.

So the idea here would be to switch from using the library like this:

```js
import SchemaViewer from 'react-schema-viewer';
```
to
```js
import {SchemaViewer} from 'react-schema-viewer';
```
or, for the Joi support,
```js
import {JoiSchemaViewer} from 'react-schema-viewer';
```

With that change, Webpack will see that `joiToJson` is only referenced in the `JoiSchemaViewer` class, and if that class is not imported, will skip importing `joi-to-json` entirely.

Note that this also involves renaming "SchemaTable" to "SchemaViewer", as @helfi92 mentioned in #21.

**Comment by arshadkazmi42** [comment]
Thank you for the detailed info. I will start working on it :+1:

**Comment by helfi92** [comment]
Published in `react-schema-viewer@3.0.0`.
